client/src/modules/dashboard/Events/components/EventFormUtils.ts

```diff
@@ -161,3 +161,83 @@ export const formatValue = (field: Field, store?: IEventData): any => {
 
   return store[key];
 };
+
+export const getAllowedSponsorTypes = (
+  sponsorData: SponsorsQuery,
+  sponsorTypes: EventSponsorTypeInput[],
+  watchSponsorsArray: EventSponsorInput[],
+  sponsorFieldId: number,
+) =>
+  sponsorTypes.filter(({ type }) =>
+    hasTypeAllowedSponsors(
+      sponsorData,
+      type,
+      watchSponsorsArray,
+      sponsorFieldId,
+    ),
+  ) ?? [];
+
+export const getAllowedSponsorsForType = (
+  sponsorData: SponsorsQuery,
+  sponsorType: string,
+  watchSponsorsArray: EventSponsorInput[],
+  sponsorFieldId?: number,
+) =>
+  sponsorData.sponsors.filter(
+    (sponsor) =>
+      hasMatchingSponsorType(sponsor, sponsorType) &&
```

Comment:
```suggestion
      sponsor.type === sponsorType &&
```
I don't think we need any abstraction here. We only use `hasMatchingSponsorType` here and it's easy enough to understand.